scripts/cloudhealth/cloudhealth.py

```diff
@@ -0,0 +1,207 @@
+
+        params:
+            account_id: id number,
+                returns list of all accounts if account_id=None
+            page: page number
+            per_page: items per page
+        '''
+        params = {'api_key':self.api_key}
+
+        if account_id:
+            endpoint = '%s/%s' % (self.endpoint, account_id)
+        else:
+            endpoint = self.endpoint
+            params.update({'page':page, 'per_page':per_page})
+
+        resp = requests.get(endpoint,
+                            params=params)
+        if resp.status_code == requests.codes.get('ok'):
+            return resp.json()
+        else:
+            resp.raise_for_status()
+
+    def _update(self, account_id, **kwargs):
+        ''' update existing accounts '''
+        params = {'api_key':self.api_key}
+        headers = {'content-type':'application/json'}
+        resp = requests.put('%s/%s' % (self.endpoint, account_id),
+                            params=params,
+                            headers=headers,
+                            json=kwargs)
+        if resp.status_code == requests.codes.get('ok'):
+            return resp.json()
+        else:
+            resp.raise_for_status()
+
+    def _delete(self, account_id=None):
+        ''' delete accounts '''
+        params = {'api_key':self.api_key}
+        resp = requests.get('%s/%s' % (self.endpoint, account_id),
+                            params=params)
+        if resp.status_code == requests.codes.get('ok'):
+            return resp.json()
+        else:
+            resp.raise_for_status()
+
+class CloudHealthAwsAccount(CloudHealthAwsAccountApi):
+    ''' class for accessing a CloudHealth AWS Account object '''
+    # pylint: disable=invalid-name
+    id = None
+    owner_id = None
+    amazon_name = None
+    authentication = {
+        #"protocol": None,
+        #"access_key": None,
+        #"secret_key": None,
+        #"assume_role_arn" : None,
+        #"assume_role_external_id" : None,
+    }
+    aws_config = {
+        #"enabled" : False,
+        #"bucket": None,
+        #"prefix": None
+    }
+    billing = {
+        #"bucket": None
+    }
+    cloudtrail = {
+        #"enabled": False,
+        #"bucket": None
+    }
+    name = None
+    tags = [
+        #{'key':None, 'value':None}
+    ]
+
+    def __init__(self, **kwargs):
+        super(CloudHealthAwsAccount, self).__init__(**kwargs)
+
+        if self.name or self.owner_id:
+            self._load()
+        else:
+            raise Exception('Missing required parameter "name".')
+
+        # pylint: disable=line-too-long
+        if 'authentication' in kwargs.keys() and \
+                'protocol' in kwargs['authentication'].keys():
+            if kwargs['authentication']['protocol'] not in ['access_key', 'assume_role']:
+                raise Exception('Protocol must be one of "access_key" or "assume_role"')
+
+            if kwargs['authentication']['protocol'] == 'access_key':
```

Comment:
Code above indicates this should only use AWS_PROFILE, not access/secret keys.  Drop support for access/secret on new tools and force AWS_PROFILE usage instead wherever it's possible.

Reply:
This is part of the Cloudhealth API, not part of the CLI invocation.